ts/src/profiler.ts

```diff
@@ -317,7 +319,17 @@ export class Profiler extends ServiceObject {
    * on the type of profile to be collected, this problem will be logged at the
    * error level and getting profile type will be retried.
    */
-  start() {
+  async start() {
```

Comment:
For my eduction: if this function async now, does it mean it implicitly returns a promise?

Reply:
Yes, this implicitly returns a `Promise<void>`.
I marked the return type.